File: domain/src/main/java/com/ahn/domain/calculator/usecase/ClearCalculatorHistoriesUseCase.kt
package com.ahn.domain.calculator.usecase

import com.ahn.domain.calculator.repository.CalculatorHistoryRepository
import javax.inject.Inject

class ClearCalculatorHistoriesUseCase @Inject constructor(
    private val repository: CalculatorHistoryRepository,
) {
    /**
     * Clears all stored calculator history entries.
     *
     * Any exceptions thrown by the underlying data layer will propagate to the caller.
     */
    suspend operator fun invoke() {
        repository.clearHistories()
    }
}

File: ios/Packages/Domain/Sources/Domain/Calculator/UseCase/ClearCalculatorHistoriesUseCase.swift
public struct ClearCalculatorHistoriesUseCase: Sendable {
    private let repository: any CalculatorHistoryRepository

    public init(repository: any CalculatorHistoryRepository) {
        self.repository = repository
    }

    public func callAsFunction() async throws {
        try await repository.clearHistories()
    }
}
